src/playlist-selectors.js

```diff
@@ -124,27 +146,32 @@ const simpleSelector = function(master, bandwidth, width, height) {
   let resolutionPlusOneSmallest = [];
   let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  stableSort(sortedPlaylists, comparePlaylistBandwidth.bind(null, useAverageBandwidth));
 
   // filter out any playlists that have been excluded due to
   // incompatible configurations or playback errors
   sortedPlaylists = sortedPlaylists.filter(Playlist.isEnabled);
   // filter out any variant that has greater effective bitrate
   // than the current estimated bandwidth
   bandwidthPlaylists = sortedPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.BANDWIDTH &&
-           elem.attributes.BANDWIDTH * Config.BANDWIDTH_VARIANCE < bandwidth;
+    const elemBandwidth = getBandwidth(elem, useAverageBandwidth);
```

Comment:
We use bandwidth comparisons for two different operations, sorting and filtering.

For sorting, if a playlist has a higher bandwidth, then it should have a higher average-bandwidth as well, so it should be safe to use `BANDWIDTH` instead of the average for sorting.

For filtering, (ignoring the filter I commented on saying the average makes a difference), we compare the bandwidths to check that they are equal for the purpose of identifying redundant streams. The attributes of redundant streams should be the same, so it should be safe to use either here. Apple also doesn't specify if the `AVERAGE-BANDWIDTH` attribute needs to be the same across redundant streams, so I may even argue not using the average is safer here to make sure primary is chosen over backup.

I'd say that it actually is less consistent with which bandwidth we are using for comparisons and guards less against missing/empty attributes. Since `AVERAGE-BANDWIDTH` is an optional property, if we opt to use it in either a sorting or filtering scenario, all it takes is for one stream in the list to not include the attribute and all the sort/filters will be incorrect.

The filter at this line however, is comparing bandwidths to the `HlsHandler`s computed bandwidth numbers in order to filter out playlists that have too high bitrate for the current bandwidth. Using the `AVERAGE-BANDWIDTH` here lets us consider a playlist we would otherwise throw out because its max bandwidth was too high, but on average, is lower bandwidth.

(made a separate comment regarding `comparePlaylistResolution`)